### DISCUSSION TITLE
Inquiry on the availability of a one-click feature to cancel all transport orders in OpenTCS

### DISCUSSION THREAD

**Discussion by BenYaho** [openTCS/opentcs#85] [Q&A] [answered]
Hi,
I am a user of OpenTCS and I am writing to inquire about the possibility of adding a one-click feature to cancel all transport orders in the system.

As you may know, canceling transport orders is an important functionality for system administrators in case of emergency or unexpected events. However, the current process of canceling each individual order is time-consuming and prone to errors. It would be very helpful if there were A simple and fast way to cancel all orders at once.

Therefore, I would like to know if OpenTCS already provides such a feature, and if not, whether it would be possible to implement it in a future release. If it is feasible, I would appreciate any suggestions or recommendations on how to achieve this functionality .

**Comment by swltr** [comment, marked as answer]
I disagree that canceling (all) transport orders is functionality you need for emergencies. In emergency situations, what you may want to do is to bring all vehicles to a halt quickly; canceling transportation processes is a different thing and does not have anything to do do with emergencies.

In the Operations Desk applications, there is already a 'pause' button (and a 'resume' button), which pauses (or resumes) all vehicles. Pausing a vehicle means it will not get any further movement commands until it is resumed. In addition to that, the vehicle driver is explicitly notified - via [onVehiclePaused()](https://www.opentcs.org/docs/5.8/developer/api-base/org/opentcs/drivers/vehicle/VehicleCommAdapter.html#onVehiclePaused(boolean)) - that the vehicle has been paused/resumed in openTCS; the vehicle driver _may_ tell the vehicle to stop immediately / resume movement as a reaction to such a notification, but that is up to the driver implementation. If implemented, clicking the 'pause' button will make all vehicles stop, and after resolving the situation/emergency, you can click the 'resume' button to continue processing the transport orders. (Alternatively, you can withdraw transport orders while the vehicles are paused.)

**Comment by BenYaho** [comment]
OK, thank you
